instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3OutputStreamWriteBytesAndOffsetAdvice.java

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0;
+
+import static io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.Injection.getInjectionObject;
+import static io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.ServletOutputStreamInjectionHelper.handleWrite;
+
+import io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet.InjectionState;
+import java.io.IOException;
+import javax.servlet.ServletOutputStream;
+import net.bytebuddy.asm.Advice;
+
+public class Servlet3OutputStreamWriteBytesAndOffsetAdvice {
+  @Advice.OnMethodEnter(skipOn = Advice.OnDefaultValue.class, suppress = Throwable.class)
+  public static boolean methodEnter(
+      @Advice.This ServletOutputStream servletOutputStream,
+      @Advice.Argument(value = 0, readOnly = false) byte[] write,
+      @Advice.Argument(value = 1, readOnly = false) int off,
+      @Advice.Argument(value = 2, readOnly = false) int len)
+      throws IOException {
+    InjectionState state = getInjectionObject(servletOutputStream);
+    // if handleWrite return true, then it means the injection has happened and the 'write'
+    // manipulate is done. the function would return false then, meaning skip the original write
```

Comment:
and include this comment in each advice
```suggestion
    // was performed. the function would return false then, meaning skip the original write
```